src/Client.php

```diff
@@ -271,26 +271,44 @@ protected function getStream($socket)
     }
 
     /**
-     * Adds an event listener to log data emitted by a stream.
+     * Generates a closure for stream output logging.
      *
-     * @param \Evenement\EventEmitter $emitter
-     * @param \Phergie\Irc\ConnectionInterface $connection Connection
-     *        corresponding to the stream
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @param string $level Evenement logging level to use
+     * @param string $prefix Prefix string for log lines (optional)
+     * @return callable
      */
-    protected function addLogging(EventEmitter $emitter, ConnectionInterface $connection)
+    protected function getOutputLogCallback(ConnectionInterface $connection, $level, $prefix = null)
     {
         $logger = $this->getLogger();
-        $callback = function($msg) use ($logger, $connection) {
-            $mask = sprintf(
-                '%s!%s@%s',
+        if (!method_exists($logger, $level)) {
+            trigger_error("Invalid log level '$level' passed to Client::getOutputLogCallback()", E_USER_ERROR);
```

Comment:
I'd suggest throwing an exception here instead of triggering an error, perhaps [`BadMethodCallException`](http://php.net/manual/en/class.badmethodcallexception.php).